version.cake

```diff
@@ -36,4 +38,78 @@ void UpdateSDKVersion(string path, string patternPrefix, string newVersion)
     ReplaceRegexInFiles(path, patternString, newString);
 }
 
+Task("UpdateCgManifestSHA").IsDependeeOf("StartNewVersion").Does(()=>
+{
+    try
+    {
+        var reposToUpdate = new List<string>(){ 
+            "https://github.com/microsoft/appcenter-sdk-android.git",
+            "https://github.com/microsoft/appcenter-sdk-apple.git",
+            "https://github.com/microsoft/appcenter-sdk-dotnet.git"
+        };
+        var manifestFilePath = "cgmanifest.json";
+        var gitHubApiPrefix = "https://api.github.com/repos";
+        var gitHubPrefix = "https://github.com";
+        var content = ParseJsonFromFile(manifestFilePath);
+        var registrations = (JArray)content["Registrations"];
+        foreach (var registration in registrations.Children())
+        {
+            var component = registration["component"];
+            if (component != null) 
+            {
+                var typeObject = component["type"];
+                if (typeObject != null && typeObject.Value<string>() == "git")
+                {
+                    var gitData = component["git"];
+                    var repoUrl = gitData["repositoryUrl"].Value<string>();
+                    if (reposToUpdate.IndexOf(repoUrl) >= 0)
+                    {
+                        var latestReleaseUrl = repoUrl.Replace(".git", "/releases/latest").Replace(gitHubPrefix, gitHubApiPrefix);
+                        var releaseJson = HttpGet(latestReleaseUrl);
+                        var releaseObject = ParseJson(releaseJson);
+                        var releaseTagObject = releaseObject["tag_name"];
+                        if (releaseTagObject != null)
+                        {
+                            var releaseTag = releaseTagObject.Value<string>();
+                            var tagsUrl = repoUrl.Replace(".git", "/tags").Replace(gitHubPrefix, gitHubApiPrefix);
+                            var tagsListJson = HttpGet(tagsUrl);
+                            var tags = JArray.Parse(tagsListJson);
+                            foreach (var tag in tags.Children())
+                            {
+                                if (tag["name"].Value<string>() == releaseTag)
+                                {
+                                    gitData["commitHash"] = tag["commit"]["sha"].Value<string>();
+                                    break;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            Warning($"Repository url: {repoUrl}. Latest release has no field 'tag_name'");
+                        }
```

Comment:
Could you please split it to multiple functions? to avoid the code like 
![image](https://user-images.githubusercontent.com/1836384/84368557-61caee00-abde-11ea-9d73-6755845321ac.png)

Reply:
Fixed